iOS/Flows/Conversation/Cells/MessagesTimeMachineCollectionViewLayout.swift

```diff
@@ -111,27 +126,45 @@ class MessagesTimeMachineCollectionViewLayout: TimeMachineCollectionViewLayout {
     /// If true, scroll to the most recent item after performing collection view updates.
     private var shouldScrollToEnd = false
     private var deletedIndexPaths: Set<IndexPath> = []
+    /// How much to adjust the proposed scroll offset.
+    private var scrollOffset: CGFloat = 0
```

Comment:
As items are inserted and deleted, we adjust this to compensate so that focused item doesn't change.